pkg/storaged/details.js

```diff
@@ -48,7 +48,24 @@
                 return client.mdraids[path].Start({ "start-degraded": { t: 'b', v: true } });
             },
             mdraid_stop: function mdraid_stop(path) {
-                return client.mdraids[path].Stop({});
+                var mdraid = client.mdraids[path];
+                var block = client.mdraids_block[path];
+                var usage = block? utils.get_usage_alerts(client, block.path) : [ ];
+
+                if (usage.length === 0)
+                    return mdraid.Stop({});
+
+                dialog.open({ Title: cockpit.format(_("Please confirm stopping of $0"),
+                                                    utils.mdraid_name(mdraid)),
```

Comment:
Shouldn't this point out that the array is currently being used? And if the user then confirms anyway, wouldn't she get the same "unexpected error" from the original issue again? Might be safer to just say that it's busy (show `usage` somewhere if that's something a human can interpret?) and ask the user to unmount it before stopping?